File: src/stores/projects.ts
import { defineStore } from "pinia";
import { ref } from "vue";
import { tauriApi, type RecentProject } from "@/services/tauri";

export const useProjectsStore = defineStore("projects", () => {
  const recent = ref<RecentProject[]>([]);
  const loading = ref(false);

  async function load() {
    loading.value = true;
    try {
      recent.value = await tauriApi.listRecentProjects();
    } finally {
      loading.value = false;
    }
  }

  async function createFromExample(example: string, targetDir: string) {
    const path = await tauriApi.seedExampleProject(targetDir, example);
    const displayName = `${example.replace(/-/g, " ")}`.replace(/\b\w/g, (c) => c.toUpperCase());
    const created = await tauriApi.addRecentProject(path, displayName);
    await load();
    return created;
  }

  async function openExisting(absPath: string, displayName: string) {
    const created = await tauriApi.addRecentProject(absPath, displayName);
    await load();
    return created;
  }

  async function remove(id: number) {
    await tauriApi.removeRecentProject(id);
    await load();
  }

  return { recent, loading, load, createFromExample, openExisting, remove };
});

File: src-tauri/src/commands/projects.rs
pub use lmt_app::projects::{
    load_project_yaml_from_path, save_project_yaml_to_path, seed_example_to_dir,
};

use lmt_shared::data::{recent_projects, Db};
use lmt_shared::dto::{ProjectConfig, RecentProject};
use lmt_shared::error::{LmtError, LmtResult};
use std::path::Path;

#[tauri::command]
pub fn load_project_yaml(abs_path: String) -> LmtResult<ProjectConfig> {
    load_project_yaml_from_path(Path::new(&abs_path))
}

#[tauri::command]
pub fn save_project_yaml(abs_path: String, config: ProjectConfig) -> LmtResult<()> {
    save_project_yaml_to_path(Path::new(&abs_path), &config)
}

#[tauri::command]
pub fn list_recent_projects(state: tauri::State<'_, Db>) -> LmtResult<Vec<RecentProject>> {
    let conn = state.lock().unwrap();
    recent_projects::list(&conn)
}

#[tauri::command]
pub fn add_recent_project(
    state: tauri::State<'_, Db>,
    abs_path: String,
    display_name: String,
) -> LmtResult<RecentProject> {
    let conn = state.lock().unwrap();
    recent_projects::upsert(&conn, &abs_path, &display_name)
}

#[tauri::command]
pub fn remove_recent_project(state: tauri::State<'_, Db>, id: i64) -> LmtResult<()> {
    let conn = state.lock().unwrap();
    recent_projects::delete(&conn, id)
}

#[tauri::command]
pub fn seed_example_project(
    app: tauri::AppHandle,
    target_dir: String,
    example: String,
) -> LmtResult<String> {
    use tauri::{Emitter, Manager};
    let resource_dir = app
        .path()
        .resource_dir()
        .map_err(|e| LmtError::Io(e.to_string()))?;
    let examples_root = resource_dir.join("examples");
    let out = seed_example_to_dir(&examples_root, &example, Path::new(&target_dir))?;
    let _ = app.emit(
        "project-seeded",
        serde_json::json!({"abs_path": out.display().to_string()}),
    );
    Ok(out.display().to_string())
}
